cmds/live.c

```diff
@@ -340,9 +340,19 @@ static int forward_options(struct uftrace_opts *opts)
 	if (capabilities < 0)
 		goto close;
 
-	status = forward_option(sfd, capabilities, UFTRACE_AGENT_OPT_PATTERN, &opts->patt_type, sizeof(opts->patt_type));
-	if (status < 0)
-		goto close;
+	if (opts->filter) { /* provide a pattern type for options that need it */
```

Comment:
`s/filter/patt_type/`?

Reply:
No, I mean to send the pattern type whenever the user uses a filter (or caller filter, trigger, args in later commits). In other words don't send the pattern type if the agent doesn't need it.

Reply:
I see, thanks for the explanation.